NEWS.md

```diff
@@ -40,6 +40,11 @@
 * `string_boundary_linter()` recognizes regular expression calls like `grepl("^abc$", x)` that can be replaced by using `==` instead (#1613, @MichaelChirico).
 * `unreachable_code_linter()` has an argument `allow_comment_regex` for customizing which "terminal" comments to exclude (#2327, @MichaelChirico). `# nolint end` comments are always excluded, as are {covr} exclusions (e.g. `# nocov end`) by default.
 * `format()` and `print()` methods for `lint` and `lints` classes get a new option `width` to control the printing width of lint messages (#1884, @MichaelChirico). The default is controlled by a new option `lintr.format_width`; if unset, no wrapping occurs (matching earlier behavior).
+* The full linter suite is roughly 14% faster due to caching of the frequently used `//SYMBOL_FUNCTION_CALL` XPath to examine function calls. (@AshesITR, #2357)
```

Comment:
this still reads like a note to ourselves rather than an announcement of a cool new feature. How about:

```
* New function node caching for big efficiency gain to most linters (e.g. up to xx%; #2357, @AshesITR). Most linters are written around usage of functions, and XPath performance searching for many functions is poor. The new `xml_find_function_calls()` entry in the `get_source_expressions()` output caches all function call nodes instead. See the vignette on creating linters for more details on how to use it. {lintr} itself saw an overall 14% improvement for running the full linter suite as a result of migrating linters to this style.
```

Reply:
Excellent. Feel free to commit that.

Reply:
I need an `xx`, could you help with that? I had in mind comparing `object_overwrite_linter()` specifically.

Reply:
Would we count the cache building time as well? `get_source_expressions()` got slower to get the speedup.

Reply:
I think `lint_package(linters = object_overwrite_linter())` is the best comparison

Reply:
7s main, 8s here. The cache really only excels in E2E comparison for many linters using it.

Reply:
Let's just drop the up to xx% then.

Reply:
Oh, duh. That linter is not a good choice -- it's only looking at `//SYMBOL`, not `//SYMBOL_FUNCTION_CALL` 🤦. How about `backport_linter()`? Maybe not a good choice since this branch makes two changes, what about comparing to the #2365 implementation?

Or maybe `unnecessary_lambda_linter()`, since that one is searching for 10s of functions, right?

Reply:
The fundamental problem is that the caching effect is a fixed cost upfront so the more linters are active the greater the improvement.

Maybe dev/compare_branches gives a better basis to reason on?

Reply:
I see, right then it's easy to cheat and find really impressive numbers but those comparisons wouldn't be fair. I think the 14% number is best, then, let me think how to re-word the NEWS.